neuralop/models/integral_transform.py

```diff
@@ -6,7 +6,47 @@
 from .mlp import MLPLinear
 
 class IntegralTransform(nn.Module):
-    def __init__(self, mlp=None,
+    """Integral Kernel Transform (GNO)
+    Computes one of the following:
+        (a) \int_{A(x)} k(x, y) dy
+        (b) \int_{A(x)} k(x, y) * f(y) dy
+        (c) \int_{A(x)} k(x, y, f(y)) dy
+        (d) \int_{A(x)} k(x, y, f(y)) * f(y) dy
+
+    x : Points for which the output is defined
+    y : Points for which the input is defined
+    A(x) : A subset of all points y (depending on 
+           each x) over which to integrate
+    k : A kernel parametrized as a MLP
+    f : Input function to integrate against given 
+        on the points y
+
+    If f is not given, a transform of type (a)
+    is computed. Otherwise transforms (b), (c),
+    or (d) are computed. The sets A(x) are specified
+    as a graph in CRS format.
+
+    Parameters
+    ----------
+    mlp : torch.nn.Module, default None
+        MLP parametrizing the kernel k. Input dimension
+        should be dim x + dim y or dim x + dim y + dim f
+    mlp_layers : list, default None
+        List of layers sizes speficing a MLP which 
+        parametrizes the kernel k. The MLP will be 
+        instansiated by the MLPLinear class  
+    mlp_non_linearity : callable, default torch.nn.functional.gelu
+        Non-linear function used to be used by the 
+        MLPLinear class. Only used if mlp_layers is 
+        given and mlp is None
+    transform_type : int, default 0
```

Comment:
Do you think a string might be less ambiguous? But even then naming is not obvious